diff --git a/lvgl/demos/widgets/lv_demo_widgets.c b/lvgl/demos/widgets/lv_demo_widgets.c
index 0d3b06837..5f9b851ff 100644
--- a/lvgl/demos/widgets/lv_demo_widgets.c
+++ b/lvgl/demos/widgets/lv_demo_widgets.c
@@ -250,6 +250,7 @@ static void profile_create(lv_obj_t * parent)
 {
     lv_obj_t * panel1 = lv_obj_create(parent);
     lv_obj_set_height(panel1, LV_SIZE_CONTENT);
+    lv_obj_set_style_radius(panel1, 0, 0);
 
     LV_IMAGE_DECLARE(img_demo_widgets_avatar);
     lv_obj_t * avatar = lv_image_create(panel1);
@@ -300,6 +301,7 @@ static void profile_create(lv_obj_t * parent)
     /*Create the second panel*/
     lv_obj_t * panel2 = lv_obj_create(parent);
     lv_obj_set_height(panel2, LV_SIZE_CONTENT);
+    lv_obj_set_style_radius(panel2, 0, 0);
 
     lv_obj_t * panel2_title = lv_label_create(panel2);
     lv_label_set_text_static(panel2_title, "Your profile");
@@ -344,6 +346,7 @@ static void profile_create(lv_obj_t * parent)
     lv_obj_t * panel3_title = lv_label_create(panel3);
     lv_label_set_text_static(panel3_title, "Your skills");
     lv_obj_add_style(panel3_title, &style_title, 0);
+    lv_obj_set_style_radius(panel3, 0, 0);
 
     lv_obj_t * experience_label = lv_label_create(panel3);
     lv_label_set_text_static(experience_label, "Experience");
