diff --git a/lvgl/src/core/lv_refr.c b/lvgl/src/core/lv_refr.c
index 989769831..814a27981 100644
--- a/lvgl/src/core/lv_refr.c
+++ b/lvgl/src/core/lv_refr.c
@@ -68,6 +68,7 @@ static void wait_for_flushing(lv_display_t * disp);
 /**********************
  *   GLOBAL FUNCTIONS
  **********************/
+extern void GPIO_1_SET(int state);
 
 /**
  * Initialize the screen refresh subsystem
@@ -582,6 +583,7 @@ static void refr_invalid_areas(void)
 {
     if(disp_refr->inv_p == 0) return;
     LV_PROFILER_REFR_BEGIN;
+    GPIO_1_SET(1);
 
     /*Find the last area which will be drawn*/
     int32_t i;
@@ -654,6 +656,7 @@ static void refr_invalid_areas(void)
 
     lv_display_send_event(disp_refr, LV_EVENT_RENDER_READY, NULL);
     disp_refr->rendering_in_progress = false;
+    GPIO_1_SET(0);
     LV_PROFILER_REFR_END;
 }
 
