diff --git a/lvgl/src/core/lv_refr.c b/lvgl/src/core/lv_refr.c
index 5bedd063b..a1a6a8831 100644
--- a/lvgl/src/core/lv_refr.c
+++ b/lvgl/src/core/lv_refr.c
@@ -46,6 +46,8 @@ static void draw_buf_flush(lv_display_t * disp);
 static void call_flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map);
 static void wait_for_flushing(lv_display_t * disp);
 
+extern void GPIO_1_SET(int state);
+
 /**********************
  *  STATIC VARIABLES
  **********************/
@@ -539,6 +541,7 @@ static void refr_invalid_areas(void)
 {
     if(disp_refr->inv_p == 0) return;
     LV_PROFILER_BEGIN;
+    GPIO_1_SET(1);
 
     /*Find the last area which will be drawn*/
     int32_t i;
@@ -568,6 +571,7 @@ static void refr_invalid_areas(void)
     }
 
     disp_refr->rendering_in_progress = false;
+    GPIO_1_SET(0);
     LV_PROFILER_END;
 }
 
